sympy/matrices/expressions/transpose.py

```diff
@@ -37,6 +37,8 @@ def doit(self, **hints):
         if hints.get('deep', True) and isinstance(arg, Basic):
             arg = arg.doit(**hints)
         try:
+            if isinstance(arg, MatrixExpr) and arg.shape == (1, 1):
+                return arg
```

Comment:
I'm not sure whether this should come here or in `MatrixExpr._eval_transpose`. Somebody else will have to comment on that. Otherwise this PR looks good.